src/config.rs

```diff
@@ -474,3 +480,22 @@ fn test_serialize_deserialize() {
     let deserialized: RawConfig = toml::from_str(&serialized).unwrap();
     assert_eq!(raw_config, deserialized);
 }
+
+#[test]
+fn test_relative_path_resolution() {
+    let mut raw_config = RawConfig::new();
+    raw_config.directories.cache_dir = Some("../cache".into());
+    raw_config.directories.custom_pages_dir = Some("../custom_pages".into());
+
+    let config_dir = Path::new("/path/to/config");
+    let config = Config::from_raw(config_dir.join("config.toml"), raw_config).unwrap();
+
+    assert_eq!(
+        config.directories.cache_dir.path(),
+        config_dir.join("../cache")
+    );
+    assert_eq!(
+        config.directories.custom_pages_dir.unwrap().path(),
+        config_dir.join("../custom_pages")
+    );
```

Comment:
I think we should replace the `config_dir.join("../something")` lines here with the absolute path we expect, for example `/path/to/cache` (or `/path/to/config/../cache`).

Reply:
Iirc that doesn't work because the path will contain `\` on Windows. We'd first need to unify the separators, which seems more fragile than doing this.

Reply:
Hm, I would think that as long as we compare `Path` objects we don't care about the separator.

In fact, I just found https://doc.rust-lang.org/std/fs/fn.canonicalize.html which we could use on the result before comparing the paths